test/tune/test_flaml_raytune_consistency.py

```diff
@@ -0,0 +1,96 @@
+        searcher = BlendSearch(
+            metric="metric", mode="min",
+            space=config_search_space,
+            low_cost_partial_config=low_cost_partial_config)
+    elif searcher_name == 'random':
+        searcher = RandomSearch(space=config_search_space)
+    else:
+        return None
+    return searcher
+
+
+def test_flaml_raytune_consistency(num_samples=-1, max_concurrent_trials=1, searcher_name='cfo'):
+    np.random.seed(100)
+    searcher = setup_searcher(searcher_name)
+    analysis = tune.run(
+        evaluate_config,    # the function to evaluate a config
+        config=config_search_space,  # the search space
+        low_cost_partial_config=low_cost_partial_config,  # a initial (partial) config with low cost
+        metric='metric',    # the name of the metric used for optimization
+        mode='min',         # the optimization mode, 'min' or 'max'
+        num_samples=num_samples,    # the maximal number of configs to try, -1 means infinite
+        time_budget_s=None,   # the time budget in seconds
+        local_dir='logs/',  # the local directory to store logs
+        search_alg=searcher,
+        # verbose=0,          # verbosity
+        # use_ray=True, # uncomment when performing parallel tuning using ray
+        )
+    flaml_best_config = analysis.best_config
+    flaml_config_in_results = [v['config'] for v in analysis.results.values()]
+    print(analysis.best_trial.last_result)  # the best trial's result
+    print('best flaml', searcher_name, flaml_best_config)  # the best config
+    print('flaml config in results', searcher_name, flaml_config_in_results)
+
+    np.random.seed(100)
+    searcher = setup_searcher(searcher_name)
+    from ray.tune.suggest import ConcurrencyLimiter
+    search_alg = ConcurrencyLimiter(searcher, max_concurrent_trials)
+    analysis = raytune.run(
+        evaluate_config,    # the function to evaluate a config
+        config=config_search_space,
+        metric='metric',    # the name of the metric used for optimization
+        mode='min',         # the optimization mode, 'min' or 'max'
+        num_samples=num_samples,    # the maximal number of configs to try, -1 means infinite
+        local_dir='logs/',  # the local directory to store logs
+        # max_concurrent_trials=max_concurrent_trials,
+        # resources_per_trial={"cpu": max_concurrent_trials, "gpu": 0},
+        search_alg=search_alg
+        )
+    ray_best_config = analysis.best_config
+    ray_config_in_results = [v['config'] for v in analysis.results.values()]
+    print(analysis.best_trial.last_result)  # the best trial's result
+    print('ray best', searcher_name, analysis.best_config)  # the best config
+    print('ray config in results', searcher_name, ray_config_in_results)
+    assert ray_best_config == flaml_best_config, 'best config should be the same'
+    assert flaml_config_in_results == ray_config_in_results, 'results from raytune and flaml should be the same'
+
+
```

Comment:
add the following function before 'if'
```suggestion
    def test_consistency():
        _test_flaml_raytune_consistency(num_samples=5, max_concurrent_trials=1, searcher_name='random')
        _test_flaml_raytune_consistency(num_samples=5, max_concurrent_trials=1, searcher_name='cfo')
        _test_flaml_raytune_consistency(num_samples=5, max_concurrent_trials=1, searcher_name='bs')
```